cpp/perspective/src/cpp/emscripten.cpp

```diff
@@ -927,18 +927,64 @@ namespace binding {
 
     template <>
     void
-    add_computed_column(std::shared_ptr<t_data_table> table, std::shared_ptr<t_data_table> flattened, const std::vector<t_rlookup>& row_indices, t_val computed_def) {
+    add_computed_column(
+        std::shared_ptr<t_data_table> table, 
+        std::shared_ptr<t_data_table> flattened,
+        const std::vector<t_rlookup>& row_indices,
+        t_val computed_def) {
         std::uint32_t end = row_indices.size();
         if (end == 0) {
             end = flattened->size();
         }
 
         t_val input_names = computed_def["inputs"];
-        std::vector<std::string> input_column_names = vecFromArray<t_val, std::string>(input_names);
-        std::string output_column_name = computed_def["column"].as<std::string>();
+        std::vector<std::string> input_column_names = 
+            vecFromArray<t_val, std::string>(input_names);
+        std::string output_column_name = 
+            computed_def["column"].as<std::string>();
         t_val type = computed_def["type"];
-        t_val computed_func = computed_def["func"];
+
+        if (has_value(computed_def["computed_function_name"])) {
+            t_computed_function_name method_name = 
+                computed_def["computed_function_name"].as<t_computed_function_name>();
+            std::vector<t_dtype> input_types;
+            std::vector<std::shared_ptr<t_column>> table_columns;
+            std::vector<std::shared_ptr<t_column>> flattened_columns;
+            for (const auto& column_name : input_column_names) {
+                auto table_column = table->get_column(column_name);
+                table_columns.push_back(table_column);
+                input_types.push_back(table_column->get_dtype());
+                flattened_columns.push_back(
+                    flattened->get_column(column_name));
+            }
+
+            // This uses the `t_computed_function` enum, not string name
+            t_computation computation = t_computed_column::get_computation(
+                method_name, input_types);
+            t_dtype output_column_type = computation.m_return_type;
+
+            // don't double create output column
+            auto schema = flattened->get_schema();
+            std::shared_ptr<t_column> output_column;
+            if (schema.has_column(output_column_name)) {
+                output_column = flattened->get_column(output_column_name);
+            } else {
+                output_column = flattened->add_column_sptr(
+                    output_column_name, output_column_type, true);
+                output_column->reserve(table_columns[0]->size());
+            }
+
+            t_computed_column::apply_computation(
+                table_columns,
+                flattened_columns,
+                output_column,
+                row_indices,
+                computation);
+            return;
+        }
         
+        // original path
+        t_val computed_func = computed_def["func"];
```

Comment:
I vote we de-support this.  It is almost certainly a better idea in any case to simply apply the function to the data before sending it to a `Table`, or use perspective's JSON API as a last-ditch effort.